### PR TITLE
added redux saga

### PR DESCRIPTION
**Pull request by ufoBeliever**
added redux saga 

closing issue https://github.com/igorizviekov/mt-booking_app/issues/25

### CODE REVIEW

File: src/components/ListingDetails/ListingDetails.tsx [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
 import { IListingDetailsProps } from './listingDetails.types';
 
 export const ListingDetails: React.FC<IListingDetailsProps> = ({
   data,
   id,
 }) => {
-  const { authData } = useAppSelector((state) => state.auth);
-
+  const { authData } = useTypedSelector((state) => state.auth);
+  console.log(authData);
```
Comment (igorizviekov) [inline src/components/ListingDetails/ListingDetails.tsx:11]: do we need that

File: src/store/types/geocodingReducer.types.d.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  licence: string;
+  powered_by: string;
+  osm_type: string;
+  osm_id: number;
+  boundingbox: {
+    '0': string;
+    '1': string;
+    '2': string;
+    '3': string;
+  };
```
Comment (igorizviekov) [inline src/store/types/geocodingReducer.types.d.ts:19]: please extract this to the separate type

Comment (igorizviekov) [reply src/store/types/geocodingReducer.types.d.ts:19]: @ufoBeliever please take action

File: src/components/ListingDetails/ListingDetails.tsx (hunk 1 above)
Comment (ufoBeliever) [reply src/components/ListingDetails/ListingDetails.tsx]: https://github.com/igorizviekov/mt-booking_app/pull/37/commits/9c3bff5628ec16686f9f6220ae73995d9479c4da

File: src/store/types/geocodingReducer.types.d.ts (hunk 2 above)
Comment (ufoBeliever) [reply src/store/types/geocodingReducer.types.d.ts]: https://github.com/igorizviekov/mt-booking_app/pull/37/commits/9c3bff5628ec16686f9f6220ae73995d9479c4da

**Review by igorizviekov — APPROVED** [review]

**Pull request merged**